src/skin/qml/qmlcontrolproxy.h

```diff
@@ -0,0 +1,89 @@
+#pragma once
+
+#include <QObject>
+#include <QQmlParserStatus>
+#include <QtQml>
+#include <memory>
+
+#include "control/controlproxy.h"
+
+namespace mixxx {
+namespace skin {
+namespace qml {
+
+class QmlControlProxy : public QObject, public QQmlParserStatus {
+    Q_OBJECT
+    Q_INTERFACES(QQmlParserStatus)
+    // The REQUIRED flag only exists in Qt 5.14 and later.
+#if QT_VERSION >= QT_VERSION_CHECK(5, 14, 0)
+    Q_PROPERTY(QString group READ getGroup WRITE setGroup NOTIFY groupChanged REQUIRED)
+    Q_PROPERTY(QString key READ getKey WRITE setKey NOTIFY keyChanged REQUIRED)
+#else
+    Q_PROPERTY(QString group READ getGroup WRITE setGroup NOTIFY groupChanged)
+    Q_PROPERTY(QString key READ getKey WRITE setKey NOTIFY keyChanged)
+#endif
+    Q_PROPERTY(QString keyValid READ isKeyValid NOTIFY keyValidChanged)
+    Q_PROPERTY(QString initialized READ isInitialized NOTIFY initializedChanged)
+    Q_PROPERTY(double value READ getValue WRITE setValue NOTIFY valueChanged)
+    Q_PROPERTY(double parameter READ getParameter WRITE setParameter NOTIFY parameterChanged)
+
+  public:
+    QmlControlProxy(QObject* parent = nullptr);
```

Comment:
I didn't want to nitpick but for the sake of safety we should better mark all one argument ctors as `explicit` to avoid any implicit conversions.

It would be nice if there was a linter rule for this kind of conventions.

Reply:
Added.